autoblocks/_impl/prompts/autogenerate.py

```diff
@@ -222,86 +208,64 @@ def generate_execution_context_class_code(prompt: Prompt) -> str:
-    auto += f"{indent()}AutoblocksPromptManager[\n"
-    auto += f"{indent(2)}{prompt.execution_context_class_name},\n"
-    auto += f"{indent(2)}{prompt.minor_version_enum_class_name},\n"
-    auto += f"{indent()}],\n"
+    auto += f"{indent()}AutoblocksPromptManager[{prompt.execution_context_class_name}],\n"
     auto += "):\n"
     auto += f'{indent()}__prompt_id__ = "{prompt.id}"\n'
     auto += f'{indent()}__prompt_major_version__ = "{prompt.major_version}"\n'
     auto += f"{indent()}__execution_context_class__ = {prompt.execution_context_class_name}\n"
     return auto + "\n"
 
 
-def generate_code_for_prompt(prompt: Prompt) -> str:
+def generate_code_for_prompt(prompt: PromptCodegen) -> str:
     return "\n\n".join(
         [
             x
             for x in [
                 generate_params_class_code(prompt),
                 generate_template_renderer_class_code(prompt),
                 generate_execution_context_class_code(prompt),
-                generate_minor_versions_enum_code(prompt),
                 generate_prompt_manager_class_code(prompt),
             ]
             if x is not None
         ],
     )
 
 
-def make_prompts_from_api_response_and_config(
-    data: List[Dict[Any, Any]],
+def make_prompts_from_config(
     config: AutogeneratePromptsConfig,
-) -> List[Prompt]:
-    # Map of prompt id to set of major versions to autogenerate code for
-    generate_for: Dict[str, Set[str]] = {}
-    for prompt in config.prompts:
-        generate_for[prompt.id] = set(prompt.major_versions)
-
-    # Map of (prompt id, major version) to (any prompt w/ that major version, list of minor versions)
-    by_major_version: Dict[Tuple[str, str], Tuple[Dict[Any, Any], List[str]]] = {}
-    for row in data:
-        prompt_id = row["id"]
-
-        if prompt_id not in generate_for:
-            continue
-
-        prompt_version = row["version"]
-
-        major_version, minor_version = prompt_version.split(".")
-
-        if major_version not in generate_for[prompt_id]:
-            continue
-
-        key = (prompt_id, major_version)
-
-        if key not in by_major_version:
-            by_major_version[key] = (row, [])
-        by_major_version[key][1].append(minor_version)
-
+) -> List[PromptCodegen]:
+    """
+    Fetches prompt data from the Autoblocks API and generates code for each prompt
+    configured in the .autoblocks.yml config file.
+    """
     prompts = []
 
-    for (prompt_id, major_version), (row, minor_versions) in by_major_version.items():
+    for prompt in config.prompts:
+        if prompt.major_version is not None:
+            major = prompt.major_version
+            minor = REVISION_LATEST
+        elif prompt.dangerously_use_undeployed_revision is not None:
+            major = REVISION_UNDEPLOYED
+            minor = prompt.dangerously_use_undeployed_revision
+        else:
+            raise ValueError("Either major_version or dangerously_use_undeployed_revision must be specified")
```

Comment:
should we just skip in this case and log rather than raising an exception?

Reply:
maybe log which prompt id too

Reply:
yeah i'll add the prompt ID but i feel it should throw right? rather than finishing and letting you discover on your own that one of your prompts is missing

Reply:
ya I think throwing makes sense thinking more about it.